ZCRMiOS/Classes/DashBoardAPIHandler.swift

```diff
@@ -0,0 +1,1737 @@
+            
+        }
+        
+    } // func ends
+    
+    
+} // extension ends
+
+
+//MARK:- HANDLER PARSING FUNCTIONS
+//MARK:-
+
+extension DashBoardAPIHandler {
+    
+    //MARK: *** DASHBOARD PARSERS ***
+    
+    internal func getDashBoardObjectFrom(_ dashBoardJSON: [String:Any] ) -> ZCRMDashBoard {
+        
+        let dashBoardObj = ZCRMDashBoard()
+        
+        for (key,value) in dashBoardJSON {
+            
+            switch (key) {
+                
+            case dashBoardAPINames.dashBoardID :
+                dashBoardObj.setDashBoard(ID: value as? String)
+                
+            case dashBoardAPINames.dashBoardName:
+                dashBoardObj.setDashBoard(Name: value as? String)
+                
+            case dashBoardAPINames.isSalesTrends:
+                dashBoardObj.setIfDashBoardIsSalesTrend(value as? Bool)
+                
+            case dashBoardAPINames.isSystemGenerated:
+                dashBoardObj.setIfDashBoardIsSystemGenerated(value as? Bool)
+                
+            case dashBoardAPINames.accessType:
+                dashBoardObj.setDashBoardAccessType(value as? String)
+                
+            case dashBoardAPINames.metaComponents:
+                let arrayOfMetaComponent = getArrayOfDashBoardMetaComponent(From: dashBoardJSON)
+                dashBoardObj.setArrayOfDashBoardMetaComponent(arrayOfMetaComponent)
+                
+            default :
+                print("UNKNOWN KEY ENCOUNTERED IN DASHBOARD PARSING !! \(key)")
+                
+            }
+            
+        }
+        
+        return dashBoardObj
+    }
+    
+} // end of extension
+
+
+
+
+extension DashBoardAPIHandler {
+    
+    
+    
+    //MARK:- *** DASHBOARD META-COMPONENT PARSERS ***
+    
+    internal func getArrayOfDashBoardMetaComponent(From dashBoardJSON:[String:Any]) -> [ZCRMDashBoardMetaComponent] {
+        
+        var metaComponentObjArray = [ZCRMDashBoardMetaComponent]()
+        
+        let metaComponentAPIName = ZCRMDashBoard.Properties.APINames.metaComponents
+        let arrayOfMetaComponentJSON = dashBoardJSON
+            .getArrayOfDictionaries(key: metaComponentAPIName)
+        
+        for dashBoardMetaComponentJSON in arrayOfMetaComponentJSON {
+            
+            let metaComponentObj = getDashBoardMetaComponentFrom(dashBoardMetaComponentJSON)
+            metaComponentObjArray.append(metaComponentObj)
+            
+        }
+        
+        return metaComponentObjArray
+    }
+    
+    
+    
+    
+    internal func getDashBoardMetaComponentFrom(_ metaComponentJSON:[String:Any]) -> ZCRMDashBoardMetaComponent {
+        
+        let metaComponentObj = ZCRMDashBoardMetaComponent()
+        
+        for (key,value) in metaComponentJSON {
+            
+            switch (key){
+                
+            case metaComponentAPINames.componentID:
+                metaComponentObj.setComponent(ID: value as? String)
+                
```

Comment:
Why didn't you use our dictionary extension?
Remove empty lines

Reply:
https://github.com/zoho/CRM-iOSSDK/pull/15#discussion_r214515160